### PR TITLE
Add cross-resource references for docdb

### LINKED ISSUE #772: DocumentDB: Add cross-resource references
Follow-up for #588 

### What problem are you facing?

The API for DocumentDB is missing cross-resource references.

### How could Crossplane help solve your problem?

The DocumentDB API should be extended to allow cross-resource references in addition to referencing the external resource directly. This includes

* `DBSubnetGroup`
* `DBClusterParameterGroup`
* `DBCluster`

### PR DESCRIPTION
**Pull request by MisterMX**
### Description of your changes

This PR adds cross-resource references for DocumentDB resources.

Fixes #772 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Manually

### CODE REVIEW

File: apis/docdb/v1alpha1/custom_types.go [hunk 1]
```diff
@@ -112,4 +121,16 @@ type CustomDBClusterParameters struct {
...
+	DBSubnetGroupNameSelector *xpv1.Selector  `json:"dbSubnetGroupNameSelector,omitempty"`
+
+	DBClusterParameterGroupNameRef      *xpv1.Reference `json:"dbClusterParameterGroupNameRef,omitempty"`
+	DBClusterParameterGroupNameSelector *xpv1.Selector  `json:"dbClusterParameterGroupNameSelector,omitempty"`
+
+	KMSKeyIDRef      *xpv1.Reference `json:"kmsKeyIDRef,omitempty"`
+	KMSKeyIDSelector *xpv1.Selector  `json:"kmsKeyIDSelector,omitempty"`
+
+	VPCSecurityGroupIDsRefs     []xpv1.Reference `json:"vpcSecurityGroupIDsRefs,omitempty"`
+	VPCSecurityGroupIDsSelector *xpv1.Selector   `json:"vpcSecurityGroupIDsSelectors,omitempty"`
```
Comment (turkenh) [inline apis/docdb/v1alpha1/custom_types.go:135]:
```suggestion
	VPCSecurityGroupIDsSelector *xpv1.Selector   `json:"vpcSecurityGroupIDsSelector,omitempty"`
```

**Review by turkenh — COMMENTED** [review]
PR for reference resolver generation got recently merged just in case you want to give it a try: https://github.com/crossplane/crossplane-tools/pull/35

Given you've already implemented and tested it, it would also be fine to merge as it is.

File: apis/docdb/v1alpha1/custom_types.go (hunk 1 above)
Comment (MisterMX) [reply apis/docdb/v1alpha1/custom_types.go]: Fixed

**Comment by MisterMX** [comment]
@turkenh thanks, I didn't know about this. I will try this the next time. However, I would like to keep the manual implementation this time if that is OK for you.

**Review by turkenh — APPROVED** [review]
> @turkenh thanks, I didn't know about this. I will try this the next time. However, I would like to keep the manual implementation this time if that is OK for you.

Sounds good. 
I would be fine to merge this once you solve the issue with CI (check-diff failing).

**Comment by MisterMX** [comment]
@turkenh cool, thanks! Error is fixed and ready to merge.

**Pull request merged**